js/ui/scheduler/ui.scheduler.recurrence_editor.js

```diff
@@ -392,10 +392,11 @@ const RecurrenceEditor = Editor.inherit({
 
     _repeatByDayValueChangeHandler() {
         let byDayRule = '';
+        const daysOffset = this._getFirstDayOfWeek();
 
         each(this._daysOfWeek, (index, day) => {
             if(day.option('value')) {
-                const dayName = days[index];
+                const dayName = days[(index + daysOffset) % 7];
```

Comment:
Can you extract method for this calculation?
It is better to use a constant instead of a number. For example, const DAYS_IN_WEEK = 7

Reply:
Done with const. Method extracting is unnecessary, because of less-productive code results.